src/compiler/utilities.ts

```diff
@@ -675,7 +674,7 @@ export function forEachEntry<K, V, U>(map: ReadonlyMap<K, V>, callback: (value:
  *
  * @internal
  */
-export function forEachKey<K, T>(map: ReadonlyCollection<K>, callback: (key: K) => T | undefined): T | undefined {
+export function forEachKey<K, T>(map: ReadonlyMap<K, unknown> | ReadonlySet<K>, callback: (key: K) => T | undefined): T | undefined {

```

Comment:
This one was funky; swap the argument types and inference breaks. Need to retest to see if my inference fixups fixed this, though.

Reply:
Nah, still screwy. I think I'll file a bug.

Reply:
#52313